src/System.Private.CoreLib/shared/System/IO/TextWriter.cs

```diff
@@ -732,11 +733,11 @@ public override void WriteLine()
             }
 
             // Not strictly necessary, but for perf reasons
-            public override void WriteLine(string value)
+            public override void WriteLine(string? value)
```

Comment:
Just for my knowledge if anyone wouldn't mind explain how this empty overrides would help, i saw the comment above, but still don't understand ...

Reply:
The base implementation this overrides does:
```C#
if (value != null) Write(value);
Write(CoreNewLineStr);
```
`Write` is then a virtual method that's called once or twice, which does:
```C#
if (value != null)
{
    Write(value.ToCharArray());
}
```
which allocates a char[] and passes that to another virtual call to Write, which does:
```C#
if (buffer != null)
{
    Write(buffer, 0, buffer.Length);
}
```
which makes another virtual call to Write, which in turn does a whole bunch of stuff:
```C#
if (buffer == null) throw new ArgumentNullException(nameof(buffer), SR.ArgumentNull_Buffer);
if (index < 0) throw new ArgumentOutOfRangeException(nameof(index), SR.ArgumentOutOfRange_NeedNonNegNum);
if (count < 0) throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
if (buffer.Length - index < count) throw new ArgumentException(SR.Argument_InvalidOffLen);

for (int i = 0; i < count; i++) Write(buffer[index + i]);
```
which makes N virtual calls to Write out each character individually.

But... none of that is necessary, as this is the "NullTextWriter", which just ignores anything written.  By overriding `WriteLine(string value)` as a nop, none of that unnecessary work is done.